core/store/src/trie/mod.rs

```diff
@@ -289,58 +289,76 @@ fn decode_children(cursor: &mut Cursor<&[u8]>) -> Result<[Option<CryptoHash>; 16
 }
 
 impl RawTrieNode {
-    fn encode_into(&self, out: &mut Vec<u8>) -> Result<(), std::io::Error> {
-        let mut cursor = Cursor::new(out);
+    /// Encodes node into a buffer and includes specified suffix at the end.
+    ///
+    /// Specifying suffix is equivalent to passing an empty suffix and later
+    /// extending the output explicitly.  The difference is that this method
+    /// will reserve appropriate amount of space in the output vector if suffix
+    /// is provided.
+    fn encode_into(&self, out: &mut Vec<u8>, suffix: &[u8]) {
```

Comment:
I noted that the only real user of `encode_into()` (https://github.com/near/nearcore/blob/master/core/store/src/trie/insert_delete.rs#L623) is always passing in an "empty" vector.  It is reusing the vector from previous calls by using `vec.clear()` so the capacity is not reset to 0.  So I wonder if this additional complexity is necessary ATM.  Unless of course you have some new use cases for this functionality.

Reply:
Good point.  It might be worth allocating the vector with maximum possible size in flatten_nodes then.